<?xml version="1.0" ?>
<launch xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="https://gist.githubusercontent.com/nalt/dfa2abc9d2e3ae4feb82ca5608090387/raw/roslaunch.xsd">
    <!-- Arguments -->
    <arg name="log_level" default="INFO" />
    <arg name="hardware_version" default="ned2"/>


    <node if="$(eval 'fake' in hardware_version)" name="niryo_robot_rpi" pkg="niryo_robot_rpi" type="fake_rpi_node.py" output="screen">
        <rosparam file="$(find niryo_robot_rpi)/config/ned/io_config.yaml"/>
        <param name="hardware_version" type="string" value="$(eval hardware_version.replace('fake_', ''))"/>
        <param name="log_level" type="string" value="$(arg log_level)"/>
    </node>

    <node if="$(eval hardware_version in ['one', 'ned'])"
          name="niryo_robot_rpi" pkg="niryo_robot_rpi" type="rpi_node.py" output="screen">
        <rosparam file="$(find niryo_robot_rpi)/config/$(arg hardware_version)/io_config.yaml"/>
        <rosparam file="$(find niryo_robot_rpi)/config/log_config.yaml"/>
        <param name="hardware_version" type="string" value="$(arg hardware_version)"/>
        <param name="log_level" type="string" value="$(arg log_level)"/>
    </node>

    <node if="$(eval hardware_version =='ned2')"
          name="niryo_robot_rpi" pkg="niryo_robot_rpi" type="rpi_node.py" output="screen" launch-prefix="">
        <rosparam file="$(find niryo_robot_rpi)/config/$(arg hardware_version)/io_config.yaml"/>
        <rosparam file="$(find niryo_robot_rpi)/config/log_config.yaml"/>
        <param name="hardware_version" type="string" value="$(arg hardware_version)"/>
        <param name="log_level" type="string" value="$(arg log_level)"/>
    </node>

</launch>
